### ISSUE TITLE
Save generated files in build folder

### ISSUE THREAD

**Issue by koral--** [pablisco/auto-module#4]
Usually generated files do not reside in the same folders as code written manually.
Generated files should be saved in `build/generated/automodule/src/main/kotlin` or something like that which is added as a custom source set. So it does not need to be gitignored.
At least allow to configure that.

**Comment by pablisco** [comment]
That's a great suggestion.

TL;DR; I'm exploring something similar 👌

I'm investigating different options for something like that. The problem with putting it in a generated folder in the build folder is that we have to compile it as the final bytecode is required for the graph to be evaluated. 

Running the compiler I found it already adds too much overhead to the script as it is now. (A min of 500ms)

The solution I want to explore is to generate a jar with bytecode using bytebuddy (or similar) to avoid the compilation cycle (which adds about 500ms to the script evaluation) completely on the pre-evaluation stage (settings.gradle.kts). 

The idea would be to attach the jar as a source to the `buildScript` when ready.

One of my concerns is whether the IDE will find the jar.

**Comment by pablisco** [comment]
On the latest version, I've moved to use a custom composite build stored in the `.gradle` folder. This means there is no need to put it on the `.gitignore` file anymore. Let me know if this solves this :)

**Comment by koral--** [comment]
Thanks, it works.
